Sources/Selections.swift

```diff
@@ -0,0 +1,52 @@
+public protocol Selection {
+}
+
+public struct Field: Selection {
+  let name: String
+  let alias: String?
+  let arguments: [String: GraphQLInputValue]?
+  
+  var responseName: String {
+    return alias ?? name
+  }
+  
+  let type: GraphQLOutputType
+  let selectionSet: [Selection]?
+  
+  public init(_ name: String, alias: String? = nil, arguments: [String: GraphQLInputValue]? = nil, type: GraphQLOutputType, selectionSet: [Selection]? = nil) {
+    self.name = name
+    self.alias = alias
+    
+    self.arguments = arguments
+    
+    self.type = type
+    self.selectionSet = selectionSet
+  }
+  
+  func cacheKey(with variables: [String: JSONEncodable]?) throws -> String {
+    if let argumentValues = try arguments?.evaluate(with: variables), !argumentValues.isEmpty {
+      let argumentsKey = orderIndependentKey(for: argumentValues)
+      return "\(name)(\(argumentsKey))"
+    } else {
+      return name
+    }
+  }
+}
+
+private func orderIndependentKey(for object: JSONObject) -> String {
```

Comment:
We should really make sure that JS key generation is order independent 😣

Luckily no one has reported any related bugs yet.